jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -94,7 +94,7 @@ public class InternalDowncallHandler {
 	/* The identifier denotes the passed-in argument is a heap segment which helps
 	 * extract the address from the heap argument's base object in native.
 	 */
-	private static final long DOWNCALL_HEAP_ARGUMENT_ID = 0x1;
+	private static final long DOWNCALL_HEAP_ARGUMENT_ID = 0x10000000;
```

Comment:
Isn't this a valid address that might be used?

Reply:
The value is not a real address but an identifier intended for the heap address which must be computed in native given there is no way to obtain the heap address in java.

Reply:
In normal case, the native address must be an real address unless it is intentionally verified with some invalid value (something like in this test suite in which the alignment value was passed to downcall). So I am OK with other options as long as the conflicts can be easily avoided in our code.

Reply:
Updated with `0xffffffff` as suggested above.

Reply:
Suggest it should be the full 64 bits: `0xFFFFFFFFFFFFFFFFL` (`0xFFFFFFFF` is a valid address on 64-bit systems).

Reply:
Updated with `0xFFFFFFFFFFFFFFFFL`.

Reply:
Longer term, I think we need a mechanism that avoids making assumptions based on a value or bit pattern. Perhaps instead of `HeapArgInfo` we should use a string that distinguishes between heap addresses and non-heap values. That string would only depend upon the signature of the foreign function to be invoked and could be shared by all users of the method handle so we can remove the expense of using `ThreadLocal`.

Reply:
>  That string would only depend upon the signature of the foreign function to be invoked and could be shared by all users of the method handle so we can remove the expense of using ThreadLocal.

For a given argument, I don't think its heap address can be shared via its signature because different heap arguments might identical type.

Reply:
I created https://github.com/eclipse-openj9/openj9/issues/19005 to follow-up on my suggestion above.